File: frontend/src/services/careerReadinessService.ts
import api from './api';
import type { CareerReadinessRequest, CareerReadinessResponse } from '../types';

export const careerReadinessService = {
  evaluateCareerReadiness: async (request: CareerReadinessRequest): Promise<CareerReadinessResponse> => {
    const response = await api.post<CareerReadinessResponse>('/ai/career-readiness', request);
    return response.data;
  },
};

File: backend/nexora-api/src/main/java/com/nexora/roadmap/service/CareerReadinessService.java
package com.nexora.roadmap.service;

import com.nexora.roadmap.dto.request.CareerReadinessRequest;
import com.nexora.roadmap.dto.response.CareerReadinessResponse;

/**
 * Service defining operations for evaluating career readiness based on target role and skills.
 */
public interface CareerReadinessService {

    /**
     * Evaluates a user's skills against a target role, calculating readiness score and retrieving recommendations.
     *
     * @param request Target role and current skills.
     * @return Evaluation details including score, level, matched/missing skills, recommendations and summary.
     */
    CareerReadinessResponse evaluateCareerReadiness(CareerReadinessRequest request);
}
